Name: Device Temperature Warnings
Version: 1.4.0-2
Author: Jason Robitaille
Description: This will check the battery temperature (which is used as a reference point) every 5 minutes and will issue a warning at 45°C and a different, more serious warning if it reaches 50°C. Both temperatures are the general maximum you should have. Higher temperatures can lead to device bricking.  (MIT license)

--- .orig/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ /usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -1881,6 +1881,8 @@
 	this.powerSource['usb'] = false;
 	this.powerSource['inductive'] = false;
 	this.powerdServiceUp = false;
+	this.batteryStatusTimer.cancel();
+	this.batteryStatusTimer = undefined;
 	return;
 },
 
@@ -1887,6 +1889,46 @@
 powerInitialize: function() {
 	Mojo.Log.info("SystemUI - Power service online!");
 	this.powerdServiceUp = true;
+	if(!this.batteryStatusTimer) {
+		this.batteryStatusTimer = window.setTimeout(this.getBatteryDetails.bind(this), 300000);
+	}
+},
+
+getBatteryDetails: function() {
+	this.batteryStatusTimer = window.setTimeout(this.getBatteryDetails.bind(this), 300000);
+	this.controller.serviceRequest('palm://com.palm.bus/signal/', {
+		method: 'addmatch',
+		parameters: {
+			"category":"/com/palm/power",
+			"method":"batteryStatus"
+		},
+		onSuccess : this.handleBatteryStatus.bind(this)
+	});
+	this.controller.serviceRequest('palm://com.palm.power/com/palm/power/', {
+		method: 'batteryStatusQuery'
+	});
+},
+
+handleBatteryStatus: function(payload) {
+	if(payload.temperature_C) {
+		if(payload.temperature_C >= 50) {
+			var bannerText = "Critical! Device temperature too high";
+			var soundClassName = this.onActiveCall ? "none" : "notifications";			
+			this.controller.showBanner({
+				messageText: bannerText,
+				icon: "/usr/lib/luna/system/luna-systemui/images/notification-small-error.png",
+				soundClass: soundClassName
+			},""); 
+		} else if(payload.temperature_C >= 45) {
+			var bannerText = "Warning: Device temperature is high";
+			var soundClassName = this.onActiveCall ? "none" : "notifications";			
+			this.controller.showBanner({
+				messageText: bannerText,
+				icon: "/usr/lib/luna/system/luna-systemui/images/notification-small-error.png",
+				soundClass: soundClassName
+			},""); 
+		}
+	}
 },
 
 /*
